registry/api/errcode/errors.go

```diff
@@ -97,9 +105,28 @@ func (e Error) Error() string {
 		e.Code.Message())
 }
 
-// Message returned the human-readable error message for this Error
-func (e Error) Message() string {
-	return e.Code.Message()
+// WithDetail will return a new Error, based on the current one, but with
+// some Detail info added
+func (e Error) WithDetail(detail interface{}) Error {
+	if err, ok := detail.(error); ok {
```

Comment:
I'm sure at the time there was a brilliant!! reason for it - but now I have no idea why :-)   will remove